JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,310 @@
+				// set Tokens as System Properties - new values to access_token and
+				// refresh_token
+				String accessToken = jObject.has("access_token") ? jObject.get("access_token").getAsString() : null;
+				String refreshToken = jObject.has("refresh_token") ? jObject.get("refresh_token").getAsString() : null;
+				if (null == accessToken || null == refreshToken) {
+					throw new Exception(
+							"Access/refresh token is null, Please verify the config.properties for proper inputs.");
+				}
+				System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, accessToken);
+				System.setProperty(OAUTH_REFRESH_TOKEN_VAR_STRING, refreshToken);
+			} catch (UnsupportedEncodingException uee) {
+				uee.printStackTrace();
+			} catch (Exception e) {
+				String res = "";
+				try {
+					BufferedInputStream in = new BufferedInputStream(connection.getErrorStream());
+					ByteArrayOutputStream buf = new ByteArrayOutputStream();
+					readResult(in, buf);
+					res = buf.toString("UTF-8");
+				} catch (Exception ex) {
+					// Skip when this happens, but print the error.
+					ex.printStackTrace();
+				}
+				throw e;
+			} finally {
+				connection.disconnect();
+			}
+		} catch (IOException unreportedex) {
+			unreportedex.printStackTrace();
+		}
+	}
+
+	// If there is no access token, Get it using password grant
+	private static void EnsureAccessToken() throws Exception {
+		try {
+			String accessToken = System.getenv(OAUTH_ACCESS_TOKEN_VAR_STRING);
+			String refreshToken = System.getenv(OAUTH_REFRESH_TOKEN_VAR_STRING);
+			if (null == accessToken || accessToken.isEmpty()) {
+				if (null == refreshToken || refreshToken.isEmpty()) {
+					// Obtain first access token and refresh Tokens
+					System.out.println("Access token is invalid/expired, trying to do refresh");
+					GetTokensByPasswordGrant();
+				} else {
+					try{
+						System.out.println("Attempting to use given refresh token.");
+						GetTokensByRefreshGrant();
+					}catch(){
+						System.out.println("Initial refresh token has expired. Getting new access and refresh tokens.");
+                    	GetTokensByPasswordGrant();
+					}
+				}
+			}else{
+				System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, accessToken);
+			}
+		} catch (Exception e) {
+			throw e;
+		}
+	}
+
+	// load the configuration properties
+	public static void loadProperties() {
+		prop = new Properties();
+		try (InputStream input = new FileInputStream("src/main/java/config.properties")) {
+			// load the properties file
+			prop.load(input);
+			// Get the connectionString parameters from properties prop
+			getcsProp(prop.getProperty("ConnectionString"));
+		} catch (Exception ex) {
+			ex.printStackTrace();
+		}
+	}
 
+	// main function, Call starts from here
+	public static void main(String[] args) {
+		try {
+			loadProperties();
+			SetUpDbForOAuth();
+			EnsureAccessToken();
+			ConnectToDatabase();
+		} catch (SQLTransientConnectionException connException) {
```

Comment:
Do you think it's necessary to catch these three SQL exceptions before the line on 312? I'm not sure, it's up to you.

Reply:
Yes, We can have a single exception handle for all these SQLExceotions. It was already written all these exception in old app. Removed now.